### ISSUE TITLE
Training code coming soon?

### ISSUE THREAD

**Issue by maderix** [TachibanaYoshino/AnimeGANv2#5]
Hello Tachibana san,
Great work and congratulations on the work on AnimeGANv2. I had good success in converting the models and running the models on Android. However the latency is still an issue, it takes about 500 ms to run a 128x128 patch of image using Tensorflow Android(I tried tflite but it increases the inference time strangely.)
I want to modify the network architecture and optimize its performance further to make it a real-time application (under 100 ms)
So to cut a long story short, are you planning to release the training code in near future? :)

Thank you.

**Comment by TachibanaYoshino** [comment]
You can refer to the training code of AnimeGAN. Since AnimeGANv2 is still in the research stage of technology sales, I am not sure whether all the code will be open sourced. And its generator network can still be compressed even smaller. Therefore, for the time being, I cannot have an accurate answer.

**Comment by TachibanaYoshino** [comment]
Currently, it is not recommended to change the generator network structure. If you still want to use it on the mobile terminal, you can refer to here: AnimeGANv2をCore MLに変換してiOSでつかう【変換済みモデルあり】 - Qiita https://qiita.com/john-rocky/items/e4c757306b4d8f0f2c36   
In addition, AnimeGANv2 has added a new loss function, which cannot be told before all open source. You can use it for fun.
But without money, I may not be able to do more research. Hahaha

**Comment by maderix** [comment]
Thanks for your reply. Let me see what I can do with your suggestion 😊

**Comment by TachibanaYoshino** [comment]
In fact, to run in real-time on a mobile terminal, a model size of several hundred Kb is a common mode. Simply compressing the network structure is not the most important choice. Model pruning is a necessary method.

**Comment by maderix** [comment]
Thanks for your reply.
Yes I've tried quantization for tflite and I managed to get the model size down to 2 MB but full integer quantization isn't  happening due to some ops like mirror padding being used in the network which are not supported by tflite.
Do you have any suggestions?
For example, can I replace mirror padding with normal padding?  
Thanks for your time 😊

**Comment by TachibanaYoshino** [comment]
Maybe you can find a solution from [TF Select](https://www.tensorflow.org/lite/guide/ops_select). Regarding normal padding, what kind of padding do you mean? You might as well try it first. If it doesn't work, I will change the padding method and retrain the weights for you.

**Comment by maderix** [comment]
Thanks I'll give it a try with the possible alternatives.

**Comment by maderix** [comment]
I meant instead of pattern mode 'reflect' being used in the generator, is it possible to use a different pattern mode in the generator layer.
I think I might figure out a solution soon . If not with modifying the current generator weights , I can probably try training a smaller model with the generator model used as a teacher for the smaller model.
I'll keep you posted regarding my efforts. 
Thank you.

**Comment by maderix** [comment]
[animeganv2-hayao-models.zip](https://github.com/TachibanaYoshino/AnimeGANv2/files/5082590/animeganv2-hayao-models.zip)
Here are the converted models from the checkpoint files if anyone's interested. :)
